### PR TITLE
**V3** Bridged CLI gap between V2 and V3

### PR DESCRIPTION
**Pull request by kunagpal**
The import command has been deferred to a later date, Newman now uses the `v2.3.1` of the `postman-runtime` library.

Also, the super sandbox collection integration test seems to be failing randomly across builds for Node `v5`.

### CODE REVIEW

**Comment by shamasis** [comment]
1. when the legacy cli flow starts, show a warning (similar to how it shows when node version mismatch occurs, but in yellow colour)

2. for all discontinued legacy switches, mention that in CLI help text.

3. for all legacy switches (not yet discontinued), mention deprecated in CLI help text.

**Comment by kunagpal** [comment]
Okay.

File: lib/cli/index.js [hunk 1]
```diff
@@ -289,6 +401,12 @@ var _ = require('lodash'),
      * @returns {*}
      */
     rawOptions = function (procArgv, program, callback) {
+        if (procArgv.indexOf('-v') > -1 || procArgv.indexOf('--version') > -1) {
```
Comment (shamasis) [inline lib/cli/index.js]: add it to parser

Comment (shamasis) [inline lib/cli/index.js]: use the parser parsed options to act.

File: bin/newman.js [hunk 2]
```diff
@@ -15,6 +16,9 @@ var _ = require('lodash'),
     dispatch = function (options, callback) {
         var command = options.command;
 
+        if (_.isString(semver.valid(options))) {
```
Comment (shamasis) [inline bin/newman.js]: @czardoz

File: lib/cli/index.js [hunk 3]
```diff
@@ -58,73 +61,160 @@ var _ = require('lodash'),
...
+                description: 'The command line collection runner for Postman',
                 debug: true, // Prevents argparse from exiting, and allows us to handle any errors.
                 prog: program
             }),
             collectionGroup = parser.addMutuallyExclusiveGroup({ required: true }),
             environmentGroup = parser.addMutuallyExclusiveGroup();
 
         collectionGroup.addArgument(['-c', '--collection'], {
-            help: 'Specify a Postman collection as a JSON file'
+            help: 'DEPRECATED: Specify a Postman collection as a JSON file'
```
Comment (czardoz) [inline lib/cli/index.js]: at the end: `(deprecated)`

File: lib/cli/index.js [hunk 4]
```diff
@@ -58,73 +61,160 @@ var _ = require('lodash'),
...
+
+        parser.addArgument(['-y', '--delay'], {
+            help: 'DEPRECATED: Specify the extent of delay between requests (milliseconds)',
+            type: Integer,
+            defaultValue: 0
         });
 
         parser.addArgument(['-R', '--avoidRedirects'], {
-            help: 'Prevents Newman from automatically following redirects',
+            help: 'DEPRECATED: Prevents Newman from automatically following redirects',
```
Comment (czardoz) [inline lib/cli/index.js]: Only the discontinued ones should be marked as such.

File: lib/cli/index.js [hunk 5]
```diff
@@ -61,160 +58,73 @@ var _ = require('lodash'),
...
-            defaultValue: false,
-            action: 'storeTrue'
-        });
-
-        parser.addArgument(['-x', '--exitCode'], {
-            help: 'DEPRECATED: Continues running tests despite test failures, but exit with code 1',
-            type: Boolean,
-            defaultValue: false,
-            action: 'storeTrue'
+            help: 'Export a HTML report to a specified file.'
```
Comment (shamasis) [inline lib/cli/index.js]: an HTML?

File: lib/run/index.js [hunk 6]
```diff
@@ -86,6 +86,11 @@ module.exports = function (options, callback) {
             globals: options.globals,
             entrypoint: options.folder,
             data: options.iterationData,
+            delay: {
+                item: options.delayRequest
+            },
+            suppressExitCode: options.suppressExitCode,
+            stopOnError: options.stopOnError,
```
Comment (czardoz) [inline lib/run/index.js]: When passing options to runtime, this should be `stopOnFailure`.

File: lib/run/index.js [hunk 7]
```diff
@@ -86,6 +86,11 @@ module.exports = function (options, callback) {
             globals: options.globals,
             entrypoint: options.folder,
             data: options.iterationData,
+            delay: {
+                item: options.delayRequest
+            },
+            suppressExitCode: options.suppressExitCode,
```
Comment (czardoz) [inline lib/run/index.js]: This will have no effect, because Runtime doesn't know anything about exit codes. It's a newman-level option :)

**Comment by czardoz** [comment]
These have been pushed to `feature/v3` :+1:

**Pull request closed without merging**